File: DomainModel/Rating.cs
﻿using System;

namespace Swk5.GeoCaching.DomainModel {
    public class Rating : IEquatable<Rating> {
        private int grade;

        public Rating(int id, int cacheId, int creatorId, DateTime creationDate, int grade) {
            Id = id;
            CacheId = cacheId;
            CreatorId = creatorId;
            CreationDate = creationDate;
            Grade = grade;
        }

        public Rating() {}

        public int Id { get; set; }
        public int CacheId { get; set; }
        public int CreatorId { get; set; }
        public DateTime CreationDate { get; set; }

        public int Grade {
            get { return grade; }
            set {
                if (value >= 1 && value <= 10) {
                    grade = value;
                }
                else {
                    throw new ArgumentException();
                }
            }
        }

        public bool Equals(Rating other) {
            if (ReferenceEquals(null, other)) {
                return false;
            }
            if (ReferenceEquals(this, other)) {
                return true;
            }
            return Id == other.Id;
        }

        public override bool Equals(object obj) {
            if (ReferenceEquals(null, obj)) {
                return false;
            }
            if (ReferenceEquals(this, obj)) {
                return true;
            }
            if (obj.GetType() != GetType()) {
                return false;
            }
            return Equals(( Rating ) obj);
        }

        public override int GetHashCode() {
            return Id;
        }
    }
}

File: GeoCachingWebClient/src/at/wea5/geocaching/domainmodel/Rating.java
package at.wea5.geocaching.domainmodel;

import java.util.Date;

public class Rating {

//------------------------------------ constructor ------------------------------------
    
    public Rating(int grade, int id, int cacheId, int creatorId, Date creationDate) {
        this.grade = grade;
        this.id = id;
        this.cacheId = cacheId;
        this.creatorId = creatorId;
        this.creationDate = creationDate;
    }
       
//-------------------------------------- public ---------------------------------------

    public int getGrade() {
        return grade;
    }
    
    public void setGrade(int grade) {
        this.grade = grade;
    }
    
    public int getId() {
        return id;
    }
    
    public void setId(int id) {
        this.id = id;
    }
    
    public int getCacheId() {
        return cacheId;
    }
    
    public void setCacheId(int cacheId) {
        this.cacheId = cacheId;
    }
    
    public int getCreatorId() {
        return creatorId;
    }
    
    public void setCreatorId(int creatorId) {
        this.creatorId = creatorId;
    }
    
    public Date getCreationDate() {
        return creationDate;
    }
    
    public void setCreationDate(Date creationDate) {
        this.creationDate = creationDate;
    }
    
//------------------------------------- private ---------------------------------------

//-------------------------------------- members --------------------------------------
 
    private int grade;
    private int id; 
    private int cacheId;
    private int creatorId;
    private Date creationDate; 
}
